File: Sche_Treatment_Project/src/Models/Doctors/DoctorResponse.ts
export class DoctorResponse {
  id: string;
  accountName: string;
  accountPhone: string;
  accountEmail: string;
  accountGender: number;
  doctorDegree: string;
  doctorRank: string;
  doctorSpecialty: string;
  doctorIntroduce: string;
  doctorExp: string;
  doctorImage: string;

  constructor(
    id: string,
    accountName: string,
    accountPhone: string,
    accountEmail: string,
    accountGender: number,
    doctorDegree: string,
    doctorRank: string,
    doctorSpecialty: string,
    doctorIntroduce: string,
    doctorExp: string,
    doctorImage: string
  ) {
    this.id = id;
    this.accountName = accountName;
    this.accountPhone = accountPhone;
    this.accountEmail = accountEmail;
    this.accountGender = accountGender;
    this.doctorDegree = doctorDegree;
    this.doctorRank = doctorRank;
    this.doctorSpecialty = doctorSpecialty;
    this.doctorIntroduce = doctorIntroduce;
    this.doctorExp = doctorExp;
    this.doctorImage = doctorImage;
  }

  getFullName(): string {
    return this.accountName;
  }

  getContactInfo(): string {
    return `Phone: ${this.accountPhone}, Email: ${this.accountEmail}`;
  }

  getNameDoctor(): string {
    return `${this.doctorRank}. ${this.doctorDegree} ${this.accountName}`;
  }
}

File: Sche_Treatment_Project_API/src/main/java/st/hcmuaf/edu/vn/sche_treatment_project_api/response/doctor/DoctorResponse.java
package st.hcmuaf.edu.vn.sche_treatment_project_api.response.doctor;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
public class DoctorResponse {
    private String id;
    private String accountName;
    private String accountPhone;
    private String accountEmail;
    private boolean accountGender;
    private String doctorDegree;
    private String doctorRank;
    private String doctorSpecialty;
    private String doctorIntroduce;
    private String doctorExp;
    private String doctorImage;
}
